test/config/mocks/i18n-loader.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+angular.module('esn.i18n')
+  .factory('esnI18nLoader', esnI18nLoader);
+
+function esnI18nLoader() {
+  var getCatalogsPromise;
+  return function(options) {
+    if (!getCatalogsPromise) {
+      getCatalogsPromise = getCatalogs();
+    }
+
+    return getCatalogsPromise
+      .then(function(catalogs) {
+        if (!catalogs[options.key]) {
+          return Promise.reject(Error('No catalog found for ' + options.key));
+        }
+
+        return catalogs[options.key];
+      });
+  };
+
+  function getCatalogs() {
+    return $qForMock.when({
```

Comment:
Why this? And it seems that you don't use this file anywhere.

Reply:
$q is not recognized is the mocked modules inside. It is recognized only test files. 

Providing it the webpack config file will actually fail the tests using $q we injected in the injector file.

So I had to provide $q with webpack as a different name and use it only in the mocked modules.

Reply:
What do you mean by `$q is not recognized is the mocked modules inside.`? Does it throw `$q is undefined` here or something? This should work fine:

```js
'use strict';

angular.module('esn.i18n')
  .factory('esnI18nLoader', esnI18nLoader);

function esnI18nLoader($q) {
  let getCatalogsPromise;

  return function(options) {
    getCatalogsPromise = getCatalogsPromise || getCatalogs();

    return getCatalogsPromise
      .then(function(catalogs) {
        if (!catalogs[options.key]) {
          return $q.reject(Error('No catalog found for ' + options.key));
        }

        return catalogs[options.key];
      });
  };

  function getCatalogs() {
    return $q.when({});
  }
}
```

Reply:
We should not use `q` - the third party library, because AngularJS doesn't understand that `q`, and you may end up having trouble with Promise not being executed when calling `$rootScope.$digest()` or such.

Reply:
The $q promises here will not resolve and result in timeouts when other tests use the mocked module.

Reply:
Let me pull your code and see what's the problem.

Reply:
This works fine without any problem:

```js
'use strict';

angular.module('esn.i18n')
  .factory('esnI18nLoader', esnI18nLoader);

function esnI18nLoader($q) {
  let getCatalogsPromise;

  return function(options) {
    if (!getCatalogsPromise) {
      getCatalogsPromise = getCatalogs();
    }

    return getCatalogsPromise
      .then(function(catalogs) {
        if (!catalogs[options.key]) {
          return $q.reject(Error('No catalog found for ' + options.key));
        }

        return catalogs[options.key];
      });
  };

  function getCatalogs() {
    return $q.when({
      en: { hello: 'hello' },
      vi: { hello: 'xin chao' }
    });
  }
}
```

![image](https://user-images.githubusercontent.com/24670327/91289746-f0eaa980-e7bc-11ea-8a82-5676bd29f400.png)

Reply:
i figured out why, the `$q` was missing for me in `esnI18nLoader($q) {` also `getCatalogs` should return something like the default catalogs.

Reply:
So ye you can remove the `q` library.

Reply:
aight, we arrived on the same thing :), thanks for help